tests/cpptests/test_cpp_iterator_not.cpp

```diff
@@ -635,3 +639,173 @@ TEST_F(NotIteratorNoChildTest, Rewind) {
+
+  // Create a mock QueryEvalCtx
+  IndexSpec *spec = (IndexSpec*)rm_calloc(1, sizeof(IndexSpec));
+  spec->rule = (SchemaRule*)rm_calloc(1, sizeof(SchemaRule));
+  spec->rule->index_all = false;
+
+  RedisSearchCtx* sctx = (RedisSearchCtx*)rm_calloc(1, sizeof(RedisSearchCtx));
+  sctx->spec = spec;
+
+  DocTable* docTable = (DocTable*)rm_calloc(1, sizeof(DocTable));
+  docTable->maxDocId = maxDocId;
+  docTable->size = maxDocId;
+
+  QueryEvalCtx* qctx = (QueryEvalCtx*)rm_calloc(1, sizeof(QueryEvalCtx));
+  qctx->sctx = sctx;
+  qctx->docTable = docTable;
+
+  QueryIterator *emptyChild = IT_V2(NewEmptyIterator)();
+  QueryIterator *it = IT_V2(NewNotIterator)(emptyChild, maxDocId, 1.0, timeout, qctx);
+
+  // Should return a wildcard iterator
+  ASSERT_EQ(it->type, WILDCARD_ITERATOR);
+  it->Free(it);
+  rm_free(qctx);
+  rm_free(docTable);
+  rm_free(sctx);
+  rm_free(spec->rule);
+  rm_free(spec);
+}
+
+TEST_F(NotIteratorReducerTest, TestNotWithEmptyChildOptimized) {
+  // Test rule 1: If the child is an empty iterator, return a wildcard iterator
+  struct timespec timeout = {LONG_MAX, 999999999};
+  t_docId maxDocId = 100;
+
+  // Create a mock QueryEvalCtx
+  IndexSpec *spec = (IndexSpec*)rm_calloc(1, sizeof(IndexSpec));
+  spec->rule = (SchemaRule*)rm_calloc(1, sizeof(SchemaRule));
+  spec->rule->index_all = true;
+  spec->existingDocs = NewInvertedIndex(Index_DocIdsOnly, 1, &spec->stats.invertedSize);
+
+  RedisSearchCtx* sctx = (RedisSearchCtx*)rm_calloc(1, sizeof(RedisSearchCtx));
+  sctx->spec = spec;
+
+  DocTable* docTable = (DocTable*)rm_calloc(1, sizeof(DocTable));
+  docTable->maxDocId = maxDocId;
+  docTable->size = maxDocId;
+
+  QueryEvalCtx* qctx = (QueryEvalCtx*)rm_calloc(1, sizeof(QueryEvalCtx));
+  qctx->sctx = sctx;
+  qctx->docTable = docTable;
+
+  QueryIterator *emptyChild = IT_V2(NewEmptyIterator)();
+  QueryIterator *it = IT_V2(NewNotIterator)(emptyChild, maxDocId, 1.0, timeout, qctx);
+
+  // Should return a wildcard iterator
+  ASSERT_EQ(it->type, READ_ITERATOR);
+  it->Free(it);
+  rm_free(qctx);
+  rm_free(docTable);
+  rm_free(sctx);
+  rm_free(spec->rule);
+  InvertedIndex_Free(spec->existingDocs);
+  rm_free(spec);
+}
+
+TEST_F(NotIteratorReducerTest, TestNotWithWildcardChild) {
+  // Test rule 2: If the child is a wildcard iterator, return an empty iterator
+  struct timespec timeout = {LONG_MAX, 999999999};
+  t_docId maxDocId = 100;
+
+  // Create a mock QueryEvalCtx
+  IndexSpec *spec = (IndexSpec*)rm_calloc(1, sizeof(IndexSpec));
+  spec->rule = (SchemaRule*)rm_calloc(1, sizeof(SchemaRule));
+  spec->rule->index_all = true;
+
+  RedisSearchCtx* sctx = (RedisSearchCtx*)rm_calloc(1, sizeof(RedisSearchCtx));
+  sctx->spec = spec;
+
+  DocTable* docTable = (DocTable*)rm_calloc(1, sizeof(DocTable));
+  docTable->maxDocId = maxDocId;
+  docTable->size = maxDocId;
+
+  QueryEvalCtx* qctx = (QueryEvalCtx*)rm_calloc(1, sizeof(QueryEvalCtx));
+  qctx->sctx = sctx;
+  qctx->docTable = docTable;
```

Comment:
Use the existing `MockQueryEvalCtx` instead
```suggestion
  std::vector<t_docId> wildcard = {1, 2, 3};
  MockQueryEvalCtx mockQctx(wildcard);
```